cli/command/cli.go

```diff
@@ -167,14 +174,23 @@ func (cli *DockerCli) RegistryClient(allowInsecure bool) registryclient.Registry
 // line flags are parsed.
 func (cli *DockerCli) Initialize(opts *cliflags.ClientOptions) error {
 	cli.configFile = cliconfig.LoadDefaultConfigFile(cli.err)
-
 	var err error
-	cli.client, err = NewAPIClientFromFlags(opts.Common, cli.configFile)
+	cli.contextStore, err = store.New(cliconfig.ContextStoreDir())
+	if err != nil {
```

Comment:
I have very mixed feelings about that: context store failing to initiate could make the CLI connect to a different API endpoint than expected quite silently

Reply:
Well, with a warning, it would not :angel: 
My main concern is here, if on a setup, there is an existing docker config but the folder (like `.docker`) is read-only (for whatever reason), before that change, it used to work, after it will fail… which doesn't seem right.

Reply:
If Simon was to move the `MkdirAll` calls from `store.New` into the place which actuall writes files (similarly to how he previously changed `New` to not write an empty `{}` file if it doesn't already exist) then this call would be read only as well (AFAICS). Would that satisfy your concern?

Reply:
Hmm, depending on the ordering of calls, root of the context store would be created either with 0700 or 0755... Is it ok with you if I make all the resources inside the context dir as 0700 ?

Reply:
Oh and there is still
```
configBytes, err := ioutil.ReadFile(configFile)
	switch {
	case os.IsNotExist(err):
		// don't do anything, currentContext is not set
	case err != nil:
		return nil, err
	default:
		var cfg config
		if err := json.Unmarshal(configBytes, &cfg); err != nil {
			return nil, err
		}
		currentContext = cfg.CurrentContext
	}
``` 
should we treat all read errors as "IsNotExist" ?

Reply:
@simonferquel with a warning (log), yeah I think so

Reply:
> Hmm, depending on the ordering of calls, root of the context store would be created either with 0700 or 0755... Is it ok with you if I make all the resources inside the context dir as 0700 ?

Either that or everywhere which makes the restricted subdir needs to first make the root of the context store with 0755, perhaps via a helper? But if the code structure (e.g. availability of the context root at various points) makes that hard then having the whole context store be 0700 doesn't seem that bad to me.

Reply:
What about removing the config.json file from the context store (which is here only to set the current context), and putting a currentContext field in the CLI config file ?

Reply:
Works for me
/cc @thaJeztah

Reply:
WFM too (and is probably less surprising for users to have it all in one place)